src/main/kotlin/org/rust/ide/refactoring/move/common/RsMoveConflictsDetector.kt

```diff
@@ -176,3 +269,22 @@ fun addVisibilityConflict(conflicts: MultiMap<PsiElement, String>, reference: Rs
     val message = "$referenceDescription uses $targetDescription which will be inaccessible after move"
     conflicts.putValue(reference, CommonRefactoringUtil.capitalize(message))
 }
+
+private val RsImplItem.implementingType: TyAdt? get() = typeReference?.type as? TyAdt
```

Comment:
What about impls for non-ADT types?

Reply:
According to [E0118](https://doc.rust-lang.org/error-index.html#E0118) inherent impl can be defined only for ADT and trait objects. So trait objects indeed are unsupported now. Impls for trait objects are quite rarely used (about 10 such impls for whole `rustc`), so I propose to add corresponding TODO comment for now